base/Base.jl

```diff
@@ -35,6 +35,7 @@ getproperty(x::Tuple, f::Int) = (@inline; getfield(x, f))
 setproperty!(x::Tuple, f::Int, v) = setfield!(x, f, v) # to get a decent error
 
 getproperty(x, f::Symbol) = (@inline; getfield(x, f))
+getproperty(f) = Base.Fix2(getproperty, f)
```

Comment:
```suggestion
getproperty(f::Symbol) = Base.Fix2(getproperty, f)
```

Reply:
I wasn't sure if we wanted to provide a type here since there are other types used for `getproperty` sometimes (e.g., `Int`, `String`).

Reply:
Are there any real-world cases (that, plus, are "widely accepted" by the community)?
`getproperty(x, f)` is supposed to be used for the syntax sugar `x.f`, not in place of `getfield(x, f)`, so `f` is almost always supposed to be `Symbol` object.

Reply:
I think the place it's most commonly used is for [DataFrames.jl](https://github.com/JuliaData/DataFrames.jl/blob/cae78ceb104d0a315d604ddca6bda43fa91a1dca/src/abstractdataframe/abstractdataframe.jl#L419).